package.json

```diff
@@ -8,16 +8,39 @@
   "engines": {
     "node": ">=16.0"
   },
-  "main": "./index.js",
+  "type": "module",
+  "main": "./dist/cjs/index.js",
+  "module": "./dist/esm/index.js",
+  "types": "./dist/types/index.d.ts",
+  "typesVersions": {
+    "*": {
+      "*": [
+        "dist/types/*.d.ts"
+      ]
+    }
+  },
   "files": [
     "**/*"
   ],
+  "exports": {
+    ".": {
+      "import": "./esm/index.js",
+      "require": "./cjs/index.js"
+    },
+    "./testing": {
+      "import": "./esm/testing/index.js",
+      "require": "./cjs/testing/index.js"
+    },
+    "./index.d.ts": "./types/index.d.ts",
+    "./package.json": "./package.json"
+  },
   "scripts": {
     "build": "run-s build:*",
     "build:0-clean": "rimraf dist coverage",
-    "build:1-compile": "tsc -p tsconfig.build.json",
-    "build:2-copy-pkg-json": "npx --yes @makerx/ts-toolkit@latest copy-package-json",
-    "build:3-copy-readme": "copyfiles README.md dist",
+    "build:1-compile-cjs": "tsc -p tsconfig.cjs.json",
+    "build:2-compile-esm": "tsc -p tsconfig.esm.json",
+    "build:3-copy-pkg-json": "npx --yes @makerx/ts-toolkit@latest copy-package-json --main ./dist/cjs/index.js --types ./dist/types/index.d.ts --custom-sections exports",
```

Comment:
### The generated `package.json` is malformed

- pointing to `./dist` for `main` and `types` top level keys.
- missing top level key `module`
- missing top level key `typesVersions`

Resolutions: Update generator or prefer top level `package.json` with `.npmignore`. The ladder would preserve `npm link` and `workspaces` functionality.

Reply:
@PhearZero good catch, should be resolved now on the latest commit, retested on generator ts repo